spikeinterface/preprocessing/filter_gaussian.py

```diff
@@ -0,0 +1,84 @@
+from typing import Iterable, Union
+import numpy as np
+from scipy.stats import norm
+from spikeinterface.core import BaseRecording, BaseRecordingSegment, get_chunk_with_margin
+from spikeinterface.core.core_tools import define_function_from_class
+from .basepreprocessor import BasePreprocessor, BasePreprocessorSegment
+
+
+class GaussianBandpassFilterRecording(BasePreprocessor):
+    """
+    Class for performing a bandpass gaussian filtering/smoothing on a recording.
+
+    Parameters
+    ----------
+    recording: BaseRecording
+        The recording extractor to be filtered.
+    freq_min: float
+        The lower frequency cutoff for the bandpass filter.
+    freq_max: float
+        The higher frequency cutoff for the bandpass filter.
+
+    Returns
+    -------
+    gaussian_bandpass_filtered_recording: GaussianBandpassFilterRecording
+        The filtered recording extractor object.
+    """
+    name = 'gaussian_bandpass_filter'
+
+    def __init__(self, recording: BaseRecording, freq_min: float = 300., freq_max: float = 5000.):
+        sf = recording.sampling_frequency
+        BasePreprocessor.__init__(self, recording)
+        self.annotate(is_filtered=True)
+
+        for parent_segment in recording._recording_segments:
+            self.add_recording_segment(GaussianFilterRecordingSegment(parent_segment, freq_min, freq_max))
+
+        self._kwargs = {'recording': recording.to_dict(), 'freq_min': freq_min, 'freq_max': freq_max}
+
+
+class GaussianFilterRecordingSegment(BasePreprocessorSegment):
+
+    def __init__(self, parent_recording_segment: BaseRecordingSegment, freq_min: float, freq_max: float):
+        BasePreprocessorSegment.__init__(self, parent_recording_segment)
+
+        self.freq_min  = freq_min
+        self.freq_max = freq_max
+
+        sf = parent_recording_segment.sampling_frequency
+        low_sigma  = sf / (2*np.pi * freq_min)
+        high_sigma = sf / (2*np.pi * freq_max)
+        self.margin = int(max(low_sigma, high_sigma) * 6. + 1)
+
+    def get_traces(self, start_frame: Union[int, None] = None, end_frame: Union[int, None] = None,
+                   channel_indices: Union[Iterable, None] = None):
+        traces, left_margin, right_margin = get_chunk_with_margin(self.parent_recording_segment, start_frame,
+                                                                  end_frame, channel_indices, self.margin)
+        dtype = traces.dtype
+        
+        traces_fft = np.fft.fft(traces, axis=0)
+        gauss_low  = self._create_gaussian(traces.shape[0], self.freq_min)
+        gauss_high = self._create_gaussian(traces.shape[0], self.freq_max)
```

Comment:
normally shape[0] should be the same at every batch when multiprocessing.
I think we should cache the gaussian for a given shape.

And if you force get_chunk_with_margin(with_zeroTrue) then normally the chunk is even the same on the border.

Reply:
that's a good suggestion!

Reply:
I don't know how multiprocessing is implemented in SI,
If a have a `dict` called `self.cached_gauss_low` that maps the shape to the gaussian, then the first process will create it and the others will be able to use it?

Reply:
No, but the same worker will run on many chunks with the same start and end frames. ;)

Reply:
You mean the same interval `end_frame - start_frame` I guess?

Thank you :)

Reply:
Yes, actually the same traces shape because of the margins

Reply:
I'm made the gaussian cache, but I'm not sure that adding zeros is correct (if I find the time I might test it).